lib/pbstrdup.c

```diff
@@ -0,0 +1,55 @@
+/* -*- c-file-style:"stroustrup"; indent-tabs-mode: nil -*- */
+#include "pbstrdup.h"
+
+#include <stdbool.h>
+#include <string.h>
+
+
+// ----------------------------------
+//        Function prototypes
+// ----------------------------------
+
+/**
+ * @brief Create a copy of specific length from provided byte-string.
+ *
+ * @param src       Pointer to the null-terminated byte string to copy.
+ * @param len       How many (at most) bytes from `src` should be written into
+ *                  copy.
+ * @param check_len Whether `len` should be checked to not be larger that `src`
+ *                  byte string length.
+ * @return Pointer to the null-terminated byte string which is copied from
+ *         `src`. `NULL` value will be returned in case of insufficient memory
+ *         error.
+ */
+static char* _pbstrndup(const char* src, size_t len, bool check_len);
```

Comment:
Question:
Do we want to document private stuff? 

I'm not against but it would be nice to have a common way how do we treat them.

Reply:
Usually, we work with private functions, and occasionally, it is not obvious what those arguments are expected to be and what the function does.

“Common” way - do you mean document all functions? I'm trying to keep to the approach where I document new code or code which I rewrite.